src/w3d/renderer/scene.cpp

```diff
@@ -0,0 +1,264 @@
+        Customized_Render(rinfo);
+    } else {
+
+        bool state = W3D::Is_Texturing_Enabled();
+#ifdef BUILD_WITH_D3D8
+        DX8Wrapper::Set_DX8_Render_State(D3DRS_ZBIAS, 0); // BFME2 calls Set_DX8_ZBias(0.0f) here
+#endif
+        Customized_Render(rinfo);
+
+        if (m_extraPassPolyRenderMode == EXTRA_PASS_LINE) {
+            W3D::Enable_Texturing(false);
+#ifdef BUILD_WITH_D3D8
+            DX8Wrapper::Set_DX8_Render_State(D3DRS_FILLMODE, D3DFILL_WIREFRAME);
+            DX8Wrapper::Set_DX8_Render_State(D3DRS_ZBIAS, 7); // BFME2 calls Set_DX8_ZBias(7.0f) here
+#endif
+            Customized_Render(rinfo);
+        } else if (m_extraPassPolyRenderMode == EXTRA_PASS_LINE) {
+            DX8Wrapper::Clear(true, false, { 0, 0, 0 }, 0.0f);
+            W3D::Enable_Texturing(false);
+#ifdef BUILD_WITH_D3D8
+            DX8Wrapper::Set_DX8_Render_State(D3DRS_FILLMODE, D3DFILL_WIREFRAME);
+            DX8Wrapper::Set_DX8_Render_State(D3DRS_ZBIAS, 7); // BFME2 calls Set_DX8_ZBias(7.0f) here
+#endif
+            Customized_Render(rinfo);
+        }
+
+        W3D::Enable_Texturing(state);
+    }
+
+    Post_Render_Processing(rinfo);
+}
+
+// 0x00832890
+SimpleSceneClass::SimpleSceneClass() : m_visibilityChecked(false) {}
+
+// 0x00832B70
+void SimpleSceneClass::Add_Render_Object(RenderObjClass *obj)
+{
+    SceneClass::Add_Render_Object(obj);
+    m_renderList.Add(obj);
+}
+
+// 0x00832BB0
+void SimpleSceneClass::Remove_Render_Object(RenderObjClass *obj)
+{
+    SceneClass::Remove_Render_Object(obj);
+    m_renderList.Remove(obj);
+}
+
+// 0x00833160
+SceneIterator *SimpleSceneClass::Create_Iterator(bool onlyvisible)
+{
+    // onlyvisible is not used!
+    return new SimpleSceneIterator(&m_renderList, onlyvisible);
+}
+
+void SimpleSceneClass::Destroy_Iterator(SceneIterator *it)
+{
+    delete it;
+}
+
+// 0x00832BF0
+void SimpleSceneClass::Register(RenderObjClass *obj, RegType for_what)
+{
+    switch (for_what) {
+        case RegType::ON_FRAME_UPDATE:
+            m_updateList.Add(obj);
+            break;
+        case RegType::LIGHT:
+            m_lightList.Add(obj);
+            break;
+        case RegType::RELEASE:
+            m_releaseList.Add(obj);
+            break;
+        default:
+            break;
+    }
+}
+
+// 0x00832C80
+void SimpleSceneClass::Unregister(RenderObjClass *obj, RegType for_what)
+{
+    switch (for_what) {
+        case RegType::ON_FRAME_UPDATE:
+            m_updateList.Remove(obj);
+            break;
+        case RegType::LIGHT:
+            m_lightList.Remove(obj);
+            break;
+        case RegType::RELEASE:
+            m_releaseList.Remove(obj);
+            break;
+        default:
+            break;
+    }
+}
+
+// 0x00832DF0
+float SimpleSceneClass::Compute_Point_Visibility(RenderInfoClass &rinfo, const Vector3 &point)
+{
+    CastResultStruct res;
+    auto camera_position = rinfo.m_camera.Get_Position();
+    LineSegClass ray{ camera_position, point };
```

Comment:
should be LineSegClass ray(camera_position, point);, if the ctor isn't there ask, ill provide the code

Reply:
It is that? do you mean `LineSegClass ray{rinfo.m_camera, point};`

Reply:
no, () there's a Vector3, Vector3 ctor